jailer.go

```diff
@@ -0,0 +1,376 @@
+func (b JailerCommandBuilder) WithDaemonize(daemonize bool) JailerCommandBuilder {
+	b.daemonize = daemonize
+	return b
+}
+
+// SecCompLevel will return the command arguments regarding secure computing
+// level.
+func (b JailerCommandBuilder) SecCompLevel() []string {
+	return []string{
+		"--seccomp-level",
+		strconv.Itoa(int(b.secCompLevel)),
+	}
+}
+
+// WithSecCompLevel will set the provided level to the builder. This represents
+// the seccomp filters that should be installed and how restrictive they should
+// be.
+func (b JailerCommandBuilder) WithSecCompLevel(level SecCompLevelValue) JailerCommandBuilder {
+	b.secCompLevel = level
+	return b
+}
+
+// Stdout will return the stdout that will be used when creating the
+// firecracker exec.Command
+func (b JailerCommandBuilder) Stdout() io.Writer {
+	return b.stdout
+}
+
+// WithStdout specifies which io.Writer to use in place of the os.Stdout in the
+// firecracker exec.Command.
+func (b JailerCommandBuilder) WithStdout(stdout io.Writer) JailerCommandBuilder {
+	b.stdout = stdout
+	return b
+}
+
+// Stderr will return the stderr that will be used when creating the
+// firecracker exec.Command
+func (b JailerCommandBuilder) Stderr() io.Writer {
+	return b.stderr
+}
+
+// WithStderr specifies which io.Writer to use in place of the os.Stderr in the
+// firecracker exec.Command.
+func (b JailerCommandBuilder) WithStderr(stderr io.Writer) JailerCommandBuilder {
+	b.stderr = stderr
+	return b
+}
+
+// Stdin will return the stdin that will be used when creating the firecracker
+// exec.Command
+func (b JailerCommandBuilder) Stdin() io.Reader {
+	return b.stdin
+}
+
+// WithStdin specifies which io.Reader to use in place of the os.Stdin in the
+// firecracker exec.Command.
+func (b JailerCommandBuilder) WithStdin(stdin io.Reader) JailerCommandBuilder {
+	b.stdin = stdin
+	return b
+}
+
+// Build will build a jailer command.
+func (b JailerCommandBuilder) Build(ctx context.Context) *exec.Cmd {
+	cmd := exec.CommandContext(
+		ctx,
+		"jailer",
+		b.Args()...,
+	)
+
+	fmt.Println("COMMAND", cmd.Args)
+
+	if stdin := b.Stdin(); stdin != nil {
+		cmd.Stdin = stdin
+	}
+
+	if stdout := b.Stdout(); stdout != nil {
+		cmd.Stdout = stdout
+	}
+
+	if stderr := b.Stderr(); stderr != nil {
+		cmd.Stderr = stderr
+	}
+
+	return cmd
+}
+
+// Jail will set up proper handlers and remove configuuration validation due to
+// stating of files
+func jail(ctx context.Context, m *Machine, cfg Config) {
+	m.Handlers.Validation = m.Handlers.Validation.Remove(ValidateCfgHandlerName)
+
+	id := StringValue(cfg.JailerCfg.ID)
+	b := JailerCommandBuilder{}.
+		WithID(id).
+		WithUID(IntValue(cfg.JailerCfg.UID)).
+		WithGID(IntValue(cfg.JailerCfg.GID)).
+		WithNumaNode(IntValue(cfg.JailerCfg.NumaNode)).
+		WithExecFile(StringValue(cfg.JailerCfg.ExecFile)).
+		WithDaemonize(cfg.JailerCfg.Daemonize).
+		WithSecCompLevel(cfg.JailerCfg.SecCompLevel).
+		WithStdout(os.Stdout).
+		WithStderr(os.Stderr)
```

Comment:
Are these all safe to use with the zero value for the type?

Reply:
These will never be nil if the validation handler is ran. So that shouldn't be a concern.

Reply:
So it is run by default (the validation handler) but is not guaranteed to be run right?

Reply:
Yea, it isn't guaranteed to run if the user explicitly turns it off. We can do a small check to see whether or not `UID` and `GID` are set and if they aren't, return an error.